Makefile

```diff
@@ -28,41 +50,51 @@ endif
 
 CONFIG_ROOT := $(shell pwd)/configs
 
-ifndef AT_CONFIGSET
-    AT_CONFIGSET := $(shell basename $$(ls -d $(CONFIG_ROOT)/[1-9]*.[0-9] | tail -1))
-    ifeq ($(AT_CONFIGSET),)
-        $(error Couldn't infer AT_CONFIGSET variable, and no hint was given... Bailing out!)
+ifndef AT_VERSION
+    AT_VERSION := $(shell basename $$(ls -d $(CONFIG_ROOT)/[1-9]*.[0-9] | tail -1))
+    ifeq ($(AT_VERSION),)
+        $(error Couldn't infer AT_VERSION variable, and no hint was given... Bailing out!)
     else
-        $(warning AT_CONFIGSET variable not informed... Using latest one ($(AT_CONFIGSET)).)
+        $(warning AT_VERSION variable not informed... Using latest one ($(AT_VERSION)).)
     endif
 endif
 
-CONFIG := $(CONFIG_ROOT)/$(AT_CONFIGSET)
+CONFIG := $(CONFIG_ROOT)/$(AT_VERSION)
 HOST_ARCH := $(shell uname -m)
-DISTRO_NAME := $(shell $(LSB_TOOL) -i -s | tr [:upper:] [:lower:])
-
-ifeq ($(HOST_ARCH),ppc64le)
-    BUILD_ARCH := $(HOST_ARCH)
-else ifeq ($(HOST_ARCH),x86_64)
-# Image for cross compiler
-    BUILD_ARCH := x86_64.ppc64le
-else
-    $(error Unsupported host architecture ($(HOST_ARCH)) to build the image)
+ifndef DISTRO_NAME
+    DISTRO_NAME := $(shell $(LSB_TOOL) -i -s | tr [:upper:] [:lower:])
+endif
+ifndef DISTRO_NICK
+    DISTRO_NICK := $(shell $(LSB_TOOL) -c -s | tr [:upper:] [:lower:])
+endif
+
+ifneq "$(HOST_ARCH)" "ppc64le"
+    ifneq "$(HOST_ARCH)" "x86_64"
+        $(error Unsupported host architecture ($(HOST_ARCH)) to build the image)
+    endif
 endif
 
 ifndef IMAGE_PROFILE
     IMAGE_PROFILE := devel
 endif
 
-DOCKER_FILE := $(CONFIG)/$(DISTRO_NAME)/Dockerfile-$(IMAGE_PROFILE)_$(BUILD_ARCH)
+DOCKER_FILE := $(CONFIG)/dockerfile-$(IMAGE_PROFILE)
+
+IMAGE_TAG := at/$(AT_VERSION):$(DISTRO_NAME)_$(IMAGE_PROFILE)_$(HOST_ARCH)
 
-IMAGE_TAG := at/$(AT_CONFIGSET):$(DISTRO_NAME)_$(IMAGE_PROFILE)_$(BUILD_ARCH)
+ifndef REPO
+    REPO := http://public.dhe.ibm.com/software/server/POWER/Linux/toolchain/at
+endif
+
+ifdef AT_EXTRA
+    EXTRA := "-0-$(AT_EXTRA)"
```

Comment:
What happens when the minor version is not `0`? e.g. AT 15.0-1-rc1

Reply:
I've added another variable: AT_MINOR